website/server/models/user/methods.js

```diff
@@ -42,6 +42,15 @@ schema.methods.addNotification = function addUserNotification (type, data = {})
   });
 };
 
+schema.methods.addNotificationUpdate = async function addUserNotificationUpdate (type, data = {}) {
+  let newNotification = {
+    type,
+    data,
+  };
+  this.notifications.push(newNotification);
```

Comment:
Yeah that was a mistake on my part. I was thinking about keeping the local version in sync with the database version and forgot that the database won't handle the deduping on a later save.